lang/cs/Org.Apache.REEF.Wake/Remote/Impl/StreamingTransportClient.cs

```diff
@@ -98,17 +97,13 @@ public void Send(T message)
         /// </summary>
         public void Dispose()
         {
-            Dispose(true);
-            GC.SuppressFinalize(this);
-        }
-
-        protected void Dispose(bool disposing)
-        {
-            if (!_disposed && disposing)
+            if (!_disposed)
             {
                 _link.Dispose();
                 _disposed = true;
             }
+
+            GC.SuppressFinalize(this);
```

Comment:
I cleaned it up at other places also and also made the StreamingLink class sealed.